src/coreclr/tools/Common/Compiler/InstructionSetSupport.cs

```diff
@@ -96,18 +96,33 @@ public SimdVectorLength GetVectorTSimdVector()
         {
             if ((_targetArchitecture == TargetArchitecture.X64) || (_targetArchitecture == TargetArchitecture.X86))
             {
-                Debug.Assert(InstructionSet.X64_AVX2 == InstructionSet.X86_AVX2);
-                Debug.Assert(InstructionSet.X64_SSE2 == InstructionSet.X86_SSE2);
-                if (IsInstructionSetSupported(InstructionSet.X86_AVX2))
+                Debug.Assert(InstructionSet.X64_VectorT512 == InstructionSet.X86_VectorT512);
+                Debug.Assert(InstructionSet.X64_VectorT256 == InstructionSet.X86_VectorT256);
+                Debug.Assert(InstructionSet.X64_VectorT128 == InstructionSet.X86_VectorT128);
+
+                if (IsInstructionSetSupported(InstructionSet.X64_VectorT256))
+                {
                     return SimdVectorLength.Vector256Bit;
-                else if (IsInstructionSetExplicitlyUnsupported(InstructionSet.X86_AVX2) && IsInstructionSetSupported(InstructionSet.X64_SSE2))
+                }
+                else if (IsInstructionSetSupported(InstructionSet.X64_VectorT128))
+                {
                     return SimdVectorLength.Vector128Bit;
+                }
                 else
+                {
                     return SimdVectorLength.None;
+                }
             }
             else if (_targetArchitecture == TargetArchitecture.ARM64)
             {
-                return SimdVectorLength.Vector128Bit;
+                if (IsInstructionSetSupported(InstructionSet.ARM64_VectorT128))
+                {
+                    return SimdVectorLength.Vector128Bit;
```

Comment:
I believe this change will cause a small regression to startup performance on ARM64 platforms.

Reply:
cc: @JulieLeeMSFT

Reply:
Are we expecting we'd make a R2R version break if/when we expose SVE instead and that its safe to assume `VectorT128` is supported?